libs/openFrameworks/utils/ofSystemUtils.cpp

```diff
@@ -601,6 +625,54 @@ LRESULT CALLBACK WndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam)
 }
 #endif
 
+bool ofSystemConfirmDialog(string message, bool bYesNo){
+#if defined( TARGET_LINUX ) && defined (OF_USING_GTK)
+	auto locale = std::locale();
+	initGTK();
+	ConfirmDialogData dialogData;
+	dialogData.message = message;
+	dialogData.yesno = bYesNo;
+	dialogData.done = false;
+	g_main_context_invoke(g_main_loop_get_context(ofGstUtils::getGstMainLoop()), &confirm_dialog_gtk, &dialogData);
+	if(!dialogData.done){
+		std::unique_lock<std::mutex> lock(dialogData.mutex);
+		dialogData.condition.wait(lock);
+	}
+	resetLocale(locale);
+	return dialogData.result;
+#endif
+
+#ifdef TARGET_OSX
+	@autoreleasepool {
+		NSAlert *alert = [[[NSAlert alloc] init] autorelease];
+		if (bYesNo) {
+			[alert addButtonWithTitle:@"Yes"];
+			[alert addButtonWithTitle:@"No"];
+		} else {
+			[alert addButtonWithTitle:@"OK"];
+			[alert addButtonWithTitle:@"Cancel"];
+		}
+		[alert setMessageText:[NSString stringWithCString:message.c_str()
+												 encoding:NSUTF8StringEncoding]];
+		NSInteger returnCode = [alert runModal];
```

Comment:
restoreAppWindowFocus();
needs to be added after this line so the ofApp regains focus.